cms/djangoapps/models/settings/course_metadata.py

```diff
@@ -408,6 +422,15 @@ def validate_proctoring_settings(cls, descriptor, settings_dict, user):
             ).format(support_email=settings.PARTNER_SUPPORT_EMAIL or 'support')
             errors.append({'key': 'proctoring_provider', 'message': message, 'model': proctoring_provider_model})
 
+        # check that a user should actually be able to update the provider to lti, which
+        # should only be allowed if the exams IDA is enabled for a course
+        available_providers = get_available_providers()
+        updated_provider = settings_dict.get('proctoring_provider', {}).get('value')
+        if updated_provider == 'lti_external' and not exams_ida_enabled(descriptor.id):
```

Comment:
Because LTI is always an available option, the above code will not trigger an invalid provider exception. Here we should actually check if the course has the exams IDA enabled if a user is trying to use the LTI provider, and create an error if the IDA is not enabled.

Reply:
nice, this alleviates my concern with courses getting stuck in a bad state if the flag is removed.